### ISSUE TITLE
switch job executor on after succesfull boot

### ISSUE THREAD

**Issue by jangalinski** [camunda/camunda-spring-boot-starter#12]
It is not a good idea to start the job executor on startup. It should only be started when the engine has been set up correctly.
So when the config requires a job executor, its startup should be postponed till the complete boot server is running.
Once such a tool is in place, it might also allow switching the executor on and off during runtime.

**Comment by osteinhauer** [comment]
We can implement a [SpringApplicationRunListener](http://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/SpringApplicationRunListener.html) and register the implementation in `src/main/resources/META-INF/spring.factories` like `org.springframework.boot.SpringApplicationRunListener=x.x.x`.

The Listener can start the job executor by injecting the bean and call the `start()` method.

To have access during runtime all you need is injecting the job executor. Another great starter is the [remote shell](http://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-remote-shell.html) which could be used.
